### PR TITLE
Add parameters to foreman::config::puppetdb

### PR DESCRIPTION
**Pull request by ekohl**
This replaces https://github.com/theforeman/puppet-foreman/pull/184 since I no longer have access to that repository.

### CODE REVIEW

**Comment by ekohl** [comment]
Rebased. Also includes #281 which should be merged first.

File: manifests/plugin/puppetdb.pp [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Installs puppetdb_foreman plugin
-class foreman::plugin::puppetdb {
+#
+# === Parameters:
+#
+# $enabled::                Whether to enable the plugin
```
Comment (domcleal) [inline manifests/plugin/puppetdb.pp]: type:boolean, and validate it

File: manifests/plugin/puppetdb.pp [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+#
+# $address::                URL of the puppetdb plugin should use to connect
```
Comment (domcleal) [inline manifests/plugin/puppetdb.pp]: "URL of the PuppetDB instance" perhaps?

File: templates/puppetdb_plugin.yaml.erb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+:puppetdb:
```
Comment (domcleal) [inline templates/puppetdb_plugin.yaml.erb]: Check the format of this file.. there's the old hash style used by the plugin which is discouraged, and there's the new style which is DB settings.  The latter would be outside of the 'puppetdb' hash, just top-level puppetdb_enabled/address.

**Comment by domcleal** [comment]
I like the look of it!

**Comment by ekohl** [comment]
Closed in favour of https://github.com/theforeman/puppet-foreman/issues/383

**Pull request closed without merging**